### ISSUE TITLE
PHOTON: `WiFi.hasCredentials()`not TRUE following `particle setup wifi`

### ISSUE THREAD

**Issue by pomplesiegel** [spark/firmware#561]
Within our program, which runs in `SYSTEM_MODE(MANUAL)`, `setup()` has a quick check 

```
//if no creds found, enter listening mode
if( !WiFi.hasCredentials() ) 
  WiFi.listen();
```

With this line in place, Photon re-enters listening mode directly following a successful call to `particle setup wifi`. 

The system LED goes green for a quick moment following a reset (as if the Photon will now attempt to connect to a network), but quickly re-enters listening mode. 

Removing the check allows successful connection to a stored network.

**Comment by pomplesiegel** [comment]
Likely related to #558

**Comment by m-mcgowan** [comment]
@pomplesiegel  can you try the workaround in 588 in your application, or alternatively build system firmware from develop, which has the fix?

**Comment by pomplesiegel** [comment]
Sure!

**Comment by pomplesiegel** [comment]
Yes, the fix from #558 fixed this issue. Closing now. Thanks!
